API/Async.pm

```diff
@@ -377,6 +395,51 @@ sub getLatestCollectionTimestamp {
 	});
 }
 
+sub updateFavorite {
+	my ($self, $cb, $action, $type, $id) = @_;
+
+	$self->getToken(sub {
+		my ($token) = @_;
+
+		if (!$token) {
+			my $error = $1 || 'NO_ACCESS_TOKEN';
+			$error = 'NO_ACCESS_TOKEN' if $error !~ /429/;
+
+			$cb->({
+				name => string('Did not get a token' . $error),
+				type => 'text'
+			});
```

Comment:
My very personal POV is that we should not have menu code in the API implementation. We should return parsed content, and leave the formatting to the caller. And that string token isn't really a token 😁.

Reply:
Candidly I just copy what you had in _get

Reply:
Really? I can't find that (but am on the phone, might miss the obvious).

Reply:
https://github.com/michaelherger/lms-plugin-tidal/blob/78ccaac499ae7c12899e21058f067e09598713ce/API/Async.pm#L412

Reply:
Oh my... 🤦🏻 I'll have to fix this.